### ISSUE TITLE
Double-check LS groups in example 7.2

### ISSUE THREAD

**Issue by juberti** [rtcweb-wg/jsep#582]
TODO: LS for offer-B1 (should omit?) and answer-B2 (should have v1?)

**Comment by juberti** [comment]
Also: remove MSIDs for recvonly/inactive m= sections.

**Comment by juberti** [comment]
Second part of this was done by #585. @fluffy to check on whether the LS groups in S 7.2 are correct.

**Comment by fluffy** [comment]
TL;DR; - summary is I think that the LS group in answer b2 needs to change from "a=group:LS a1" to "a=group:LS a1 v1"

Long version 

The relevant part of the offer is

a=group:LS a1 v1

m=audio 
a=mid:a1
a=sendrecv
a=msid:71317484-2ed4-49d7-9eb7-1414322a7aae
       6a7b57b8-f043-4bd1-a45d-09d4dfa31226

m=video 
a=mid:v1
a=sendrecv
a=msid:71317484-2ed4-49d7-9eb7-1414322a7aae
       5ea4d4a1-2fda-4511-a9cc-1b32c2e59552

And the relevant part of the answer is  

a=group:LS a1

m=audio 
a=mid:a1
a=sendrecv
a=msid:57017fee-b6c1-4162-929c-a25110252400
       e83006c5-a0ff-4e0a-9ed9-d3e6747be7d9

m=video 
a=mid:v1
a=recvonly


Note that because the answer is recvonly for video, there is no MSID for the video in the answer as it is recv only and note the first half of the msid, which indicate the stream are the same for a1 and v1 in the offer. The question is, given that v1 is recvonly in the answer, is it included in the LS group or not ? 


The text that determines this is 4th para in 5.3.1 of jsep which says:

>  The next step is to generate session-level lip sync groups as defined in [RFC5888], Section 7. For each group of type "LS" present in the offer, determine which of the local RtpTransceivers identified by that group's mid values reference a common local MediaStream (as specified in the addTrack and addTransceiver methods). If at least two such RtpTransceivers exist, a group of type "LS" with the mid values of these RtpTransceivers MUST be added. Otherwise, this indicates a difference of opinion between the offerer and answerer regarding lip sync status, and as such, the offered group MUST be ignored and no corresponding "LS" group generated.


It just talks about RtpTransceivers and given that the transceiver for v1 has the same stream as a1, then the LS group for them should the same. The text from the LS https://tools.ietf.org/html/rfc5888#section-9.2 says 

   A SIP entity that receives an offer that contains an "a=group" line
   with semantics that are understood MUST return an answer that
   contains an "a=group" line with the same semantics.


So assuming that the thing creating the answer is doing lip sync on th the incoming a1 and v1, and it should be because the are in the same stream, the answer should look like

a=group:LS a1 v1

**Comment by juberti** [comment]
I don't think Alice has a common local MediaStream in this case though. She is going to get offer-B2 with LS a1 v1, and find that her MediaStream only includes an audio track.

Yes, Bob wants to group his tracks into a single MediaStream, but it's not clear that this will cause Alice to generate LS a1 v1 in her answer.

**Comment by fluffy** [comment]
So when offer-b2 arrives, and the v1 Track is created, it seems like the MSID would cause it to be connected to the same MediaSteam as a1 due the first half of MSID being the same ?

**Comment by juberti** [comment]
Discussed this /ad nauseam/ in SFO and decided that in the case of a=recvonly m= sections in answers, they should still be included in the answer LS group.

e.g.

offer: a1 sendrecv, v1 sendrecv, LS group = "a1 v1"
answer: a1 sendrecv, v1 recvonly, LS group = "a1 v1"

whereas the current alg would strip the LS group from the answer.

**Comment by juberti** [comment]
So, referencing the OP:
- offer-B1 should not include a LS group (no 1-item LS groups)
- answer-B2 should include v1 (per the preceding comment)
